dumux-precice/couplingadapter.cc

```diff
@@ -208,12 +208,37 @@ bool CouplingAdapter::requiresToWriteInitialData()
 bool CouplingAdapter::requiresToReadCheckpoint()
 {
     assert(wasCreated_);
+    assert(checkpointingInitialized_);
     return precice_->requiresReadingCheckpoint();
 }
 
 bool CouplingAdapter::requiresToWriteCheckpoint()
 {
     assert(wasCreated_);
+    assert(checkpointingInitialized_);
     return precice_->requiresWritingCheckpoint();
 }
+
+bool CouplingAdapter::writeCheckpointIfRequired()
+{
+    if (requiresToWriteCheckpoint() && !states_.empty()) {
+        for (auto &state : states_) {
+            state->writeState();
+        }
+        return true;
+    }
+    return false;
```

Comment:
More a question of style, but you could use an early exit here.
The loop will also do nothing if there are no states.

Same for readCheckpoint

```suggestion
    if (!requiresToWriteCheckpoint()) {
      return false;
    }
    for (auto &state : states_) {
      state->writeState();
    }
    return true;
```

Reply:
Changed :)